### ISSUE TITLE
[java] DontUseFloatTypeForLoopIndices false-negative if variable is declared before loop

### ISSUE THREAD

**Issue by csrma** [pmd/pmd#5878]
**Affects PMD Version:** 7.15.0

**Rule:** DontUseFloatTypeForLoopIndices

Please provide the rule name and a link to the rule documentation: [DontUseFloatTypeForLoopIndices](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#dontusefloattypeforloopindices)

**Description:**
The rule seems to **not detect** cases where the iteration variable (`i`) is declared before the loop. From the XPath expression, it seems the rule can detect if the variable is declared in the `ForInit` part only. But the `float i` variable might be declared before the for loop for valid reasons. For example, if the variable is needed after the loop (e.g., to check how many steps the loop ran).

**Code Sample demonstrating the issue:**

```java
class BuggyLoop {
    public static void main(String[] args) {
        float i = 0.0f;
        for (; i < 10.0f; i += 0.5f) {    // should raise DontUseFloatTypeForLoopIndices
            System.out.println(i);
            if (/*something happens*/)
                break;
        }
        System.out.println("Loop completed at level: " + i);
    }
}
```

**Expected outcome:**

PMD should report a violation at line 4, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*
